src/test/uint256_tests.cpp

```diff
@@ -386,6 +386,33 @@ BOOST_AUTO_TEST_CASE(from_hex)
     TestFromHex<Wtxid>();
 }
 
+BOOST_AUTO_TEST_CASE(from_user_hex)
+{
+    constexpr std::string_view valid_hex_65{"0x0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0"}; // 0x prefix and 65 hex chars
+    static_assert(valid_hex_65.size() == 2 + 65);
+
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("").value(), uint256::ZERO);
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("0x").value(), uint256::ZERO);
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("0").value(), uint256::ZERO);
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("0x10").value(), uint256{16});
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("10").value(), uint256{16});
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("0xff").value(), uint256{255});
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("ff").value(), uint256{255});
```

Comment:
```suggestion
    BOOST_CHECK_EQUAL(uint256::FromUserHex("0x10").value(), uint256{0x10});
    BOOST_CHECK_EQUAL(uint256::FromUserHex("10").value(), uint256{0x10});
    BOOST_CHECK_EQUAL(uint256::FromUserHex("0xff").value(), uint256{0xff});
    BOOST_CHECK_EQUAL(uint256::FromUserHex("ff").value(), uint256{0xff});
```

Reply:
Thanks, that makes much more sense indeed, taken.